### PR TITLE
replace FOREACH in spinoffs

### PR DESCRIPTION
**Pull request by white-haired-uncle**
Looks like wmlindent had a lot to do here.  The only changes I made were to replace the FOREACH loops with for/foreach, and I changed the loop variables in a couple cases that used i for both inner and outer loops (not wrong as I understand it, but IMO it's a lot more readable to just use unique and potentially less error-prone).

I haven't tested any of these.  I'm going to do some other cleanup stuff (have these been brought up to 1.16?  a recent forum comment said no) and then play test.

Ref #621

### CODE REVIEW

**Comment by Dugy** [comment]
Can you please test at least a little of some parts? It's not possible to review it because wmllindent has changed everything.

File: spinoffs/The_Beautiful_Child/utils/utils.cfg [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#textdomain wesnoth-tbc
+#textdomain wesnth-tbc
```
Comment (edwardspec) [inline spinoffs/The_Beautiful_Child/utils/utils.cfg:1]: typo

File: spinoffs/The_Beautiful_Child/utils/global_events.cfg [hunk 2]
```diff
@@ -1,51 +1,53 @@
-#textdomain wesnoth-tbc
+#textdomain wesnth-tbc
```
Comment (edwardspec) [inline spinoffs/The_Beautiful_Child/utils/global_events.cfg:1]: typo

File: spinoffs/The_Beautiful_Child/units/Bringer_of_Light2.cfg [hunk 3]
```diff
@@ -1,20 +1,21 @@
-#textdomain wesnoth-tbc
+#textd
+main wesnoth-tbc
```
Comment (edwardspec) [inline spinoffs/The_Beautiful_Child/units/Bringer_of_Light2.cfg:2]: typo

**Review by edwardspec — CHANGES_REQUESTED** [review]

**Comment by edwardspec** [comment]
> It's not possible to review it because wmllindent has changed everything.

Can add `?w=1` to the URL of GitHub diff to ignore whitespace changes: https://github.com/Dugy/Legend_of_the_Invincibles/pull/622/files?w=1

**Comment by Dugy** [comment]
Oh thanks. It looks okay to me. Except for the textdomain issues.

File: spinoffs/The_Beautiful_Child/utils/utils.cfg (hunk 1 above)
Comment (white-haired-uncle) [reply spinoffs/The_Beautiful_Child/utils/utils.cfg:1]: done

File: spinoffs/The_Beautiful_Child/utils/global_events.cfg (hunk 2 above)
Comment (white-haired-uncle) [reply spinoffs/The_Beautiful_Child/utils/global_events.cfg:1]: done

File: spinoffs/The_Beautiful_Child/units/Bringer_of_Light2.cfg (hunk 3 above)
Comment (white-haired-uncle) [reply spinoffs/The_Beautiful_Child/units/Bringer_of_Light2.cfg]: done

**Review by edwardspec — APPROVED** [review]

File: spinoffs/The_Beautiful_Child/utils/utils.cfg (hunk 1 above)
Comment (edwardspec) [reply spinoffs/The_Beautiful_Child/utils/utils.cfg:1]: Still exists.

**Comment by white-haired-uncle** [comment]
In Kill the King, it always says there's an early finish bonus, but it's 0 per turn.  I think this  is because gold/village and base income are 0.

**Comment by white-haired-uncle** [comment]
> Can you please test at least a little of some parts? It's not possible to review it because wmllindent has changed everything.

I played all of KtK, not realizing the only foreach was at the very end of the last scenario.  And it was broken ([do] before array=).

**Pull request merged**